### DISCUSSION TITLE
How to upgrade argo workflows?

### DISCUSSION THREAD

**Discussion by static-moonlight** [argoproj/argo-workflows#12834] [Q&A] [answered]
I am using the cluster installation of Argo Workflows, meaning I have one single Argo instance running in my Kubernetes cluster. I was wondering how to upgrade it properly. I am under the impression, that a clean upgrade requires a complete removal of Argo Workflows from the cluster and re-deploy it again, using the new version. As far as I know, that requires to tear down all the workflows as well, because Argo's CRDs can't be removed, when they are used by workflows. And that implies I have to tear down all workflows, tear down Argo, re-deploy Argo, and finally re-reploy all workflows again. But that doesn't sound right either.

So, what is the best approach here? How are you doing it?

**Comment by agilgur5** [comment, marked as answer]
> I am under the impression, that a clean upgrade requires a complete removal of Argo Workflows from the cluster and re-deploy it again

No, you don't need to do that.
You can just deploy the new manifests or do a `helm upgrade` if you're using [the Helm Chart](https://github.com/argoproj/argo-helm/tree/main/charts/argo-workflows). With Argo CD or other GitOps tools, you can similarly update the version and then `sync`.

That will update all manifests to the new versions, including CRDs as well as all `Deployments` to new images (etc). The `Deployment` updates will trigger a rolling deploy of the Pods (depending on your [`strategy`](https://kubernetes.io/docs/concepts/workloads/controllers/deployment/#strategy) configuration) and when that completes you'll be on the new version.
